### PR TITLE
feat(eks-deploy): define concurrency on workflow

### PR DESCRIPTION
**Pull request by vixus0**
Instead of having to configure concurrency everywhere this workflow is
used, we can define it on the workflow itself, scoped based on the
workflow name and deploy environment. This is to ensure we don't deploy
different refs to the same environment at a time.

### CODE REVIEW

File: .github/workflows/eks-deploy.yaml [hunk 1]
```diff
@@ -42,6 +42,10 @@ on:
       PRIVATE_REPO_TOKEN:
         required: true
 
+concurrency:
+  group: ${{ github.workflow }}:${{ inputs.environment || 'none' }}
+  cancel-in-progress: false
```
Comment (lillapetri) [inline .github/workflows/eks-deploy.yaml:47]: Are we sure we don't want to allow canceling?

Comment (vixus0) [reply .github/workflows/eks-deploy.yaml:47]: The workflow can still be manually cancelled, but I can't think of a situation where you'd want to automatically clobber an ongoing deployment 🤔

Comment (vixus0) [reply .github/workflows/eks-deploy.yaml:47]: (That's how the Concourse deployments work too.)

Comment (lillapetri) [reply .github/workflows/eks-deploy.yaml:47]: Ah ok, I thought it's for manual cancellation, sorry.

**Review by lillapetri — APPROVED** [review]

**Review by vinimpv — APPROVED** [review]

**Pull request merged**